app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala

```diff
@@ -91,18 +92,27 @@ class BusinessEntityController @Inject() (
   def postAddBusinessEntity: Action[AnyContent] = withCDSAuthenticatedUser.async { implicit request =>
     implicit val eori: EORI = request.eoriNumber
 
-    def handleValidAnswer(form: FormValues) =
+    def handleValidAnswer(form: FormValues, undertaking: Undertaking) =
       if (form.value === "true")
         store.update[BusinessEntityJourney](_.setAddBusiness(form.value.toBoolean)).flatMap(_.next)
-      else Redirect(routes.AccountController.getAccountPage()).toFuture
+      else {
+        val result: OptionT[Future, Result] = for {
+          uj <- store.get[UndertakingJourney].toContext
```

Comment:
NABD - I'd just called this 'journey'.

Reply:
I think it's a undertaking journey under BusinessEntityController, so better not just call it journey

Reply:
There's an argument for it being implicitly that as it's in the BusinessEntityController, and 'journey' is clearer than 'uj'. But it's so minor I don't really mind.